system_tests/espresso_migration_test.go

```diff
@@ -0,0 +1,351 @@
+	Require(t, err)
+
+	proxyAdmin, err := mocksgen.NewProxyAdminForBinding(proxyAdminAddr, l1Client)
+	Require(t, err)
+
+	callOpts := builder.L1Info.GetDefaultCallOpts("RollupOwner", ctx)
+
+	challengeManagerImplAddr, err := proxyAdmin.GetProxyImplementation(callOpts, challengeManagerAddress)
+	Require(t, err)
+	proxyAdminCallData, err := proxyAdminAbi.Pack("upgradeAndCall", challengeManagerAddress, challengeManagerImplAddr, callData) // change 2nd chall addr to actual impl addr
+	Require(t, err)
+
+	upgradeExecutor, err := upgrade_executorgen.NewUpgradeExecutor(upgradeExecutorAddress, l1Client)
+
+	Require(t, err)
+
+	tx, err := upgradeExecutor.ExecuteCall(auth, proxyAdminAddr, proxyAdminCallData)
+	Require(t, err)
+
+	_, err = builder.L1.EnsureTxSucceeded(tx)
+	Require(t, err)
+
+	checkPostUpgradeAssertions(t, proxyAdmin, callOpts, challengeManagerAddress, challengeManagerImplAddr, challengeManager, oldOspEntryAddress, newOspEntry)
+
+	log.Info("Successfully upgraded smart contracts for espresso sequencing.")
+}
+
+func TestEspressoMigration(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	builder, l2Node, l2Info, _, cleanup := BuildNonEspressoNetwork(ctx, t)
+	defer cleanup()
+	node := builder.L2
+
+	// Wait for the initial message
+	expected := arbutil.MessageIndex(1)
+	err := waitFor(t, ctx, func() bool {
+		msgCnt, err := l2Node.ConsensusNode.TxStreamer.GetMessageCount()
+		if err != nil {
+			panic(err)
+		}
+
+		validatedCnt := node.ConsensusNode.BlockValidator.Validated(t)
+		return msgCnt >= expected && validatedCnt >= expected
+	})
+	Require(t, err)
+
+	auth := builder.L1Info.GetDefaultTransactOpts("RollupOwner", ctx)
+
+	preEspressoAccount := "preEspressoAccount"
+	err = checkTransferTxOnL2(t, ctx, l2Node, preEspressoAccount, l2Info)
+	Require(t, err)
+	// Check if the tx is executed correctly
+	err = checkTransferTxOnL2(t, ctx, l2Node, "User10", l2Info)
+	Require(t, err)
+
+	// Remember the number of messages
+	var msgCnt arbutil.MessageIndex
+	err = waitFor(t, ctx, func() bool {
+		cnt, err := node.ConsensusNode.TxStreamer.GetMessageCount()
+		Require(t, err)
+		msgCnt = cnt
+		log.Info("waiting for message count", "cnt", msgCnt)
+		return msgCnt > 2
+	})
+	Require(t, err)
+
+	// Wait for the number of validated messages to catch up
+	err = waitForWith(t, ctx, 360*time.Second, 5*time.Second, func() bool {
+		validatedCnt := node.ConsensusNode.BlockValidator.Validated(t)
+		log.Info("waiting for validation", "validatedCnt", validatedCnt, "msgCnt", msgCnt)
+		return validatedCnt >= msgCnt
+	})
+	Require(t, err)
+
+	log.Info("Turn on espresso network after sequencing some default transactions.")
+
+	log.Info("Update relevant contracts on L1 to prepare for espresso sequencing.")
+
+	upgradeContracts(t, &auth, ctx, builder)
+
+	log.Info("Successfully upgraded the challenge manager to point to the new OSP entry.")
+
+	log.Info("Turn on espresso sequencing by changing the builders configs.")
```

Comment:
I doubt that we can do such a thing. In the `switch sequencer` or `espresso sequencer`, we didn't add any listener to the chain config. That means updating the chain config on L1 won't help.

I think we should turn off the L2 node (which is actually a sequencer here) and run another node to complete the migration.